contrib/platform/src/com/sun/jna/platform/unix/LibCAPI.java

```diff
@@ -223,18 +200,236 @@ public ssize_t(long value) {
+
+    // Low-Level I/O:
+
+
+    /**
+     * Creates and returns a new file descriptor for the file named by filename.
+     * Initially, the file position indicator for the file is at the beginning of the file.
+     * The argument mode is used only when a file is created, but it doesn't hurt to supply the argument in any case.
+     * <p>
+     * If on a 32 bit machine the sources are translated with _FILE_OFFSET_BITS == 64, returns
+     * a file descriptor opened in the large file mode which enables the file handling functions
+     * to use files up to 2^63 bytes in size and offset from -2^63 to 2^63.
+     * This happens transparently for the user since all the low-level file handling functions are equally replaced.
+     * <p>
+     * TODO cancellation handlers?
+     * This function is a cancellation point in multi-threaded programs.
+     * This is a problem if the thread allocates some resources (like memory, file descriptors, semaphores or whatever) at the time open is called.
+     * If the thread gets canceled these resources stay allocated until the program ends.
+     * To avoid this calls to open should be protected using cancellation handlers.
+     * <p>
+     * The open function is the underlying primitive for the TODO: fopen and freopen functions , that create streams.
+     * @param filename the name of the file to be opened.
+     * @param flags controls how the file is to be opened.
+     *             This is a bit mask; you create the value by the bitwise OR of the appropriate parameters.
+     * @return a non-negative integer file descriptor.
+     * In the case of an error, a value of -1 is returned instead.
+     * In addition to the usual file name errors, the following errno error conditions are defined for this function:
+     * {@link ErrNo#EACCES}, {@link ErrNo#EEXIST}, {@link ErrNo#EINTR}, {@link ErrNo#EISDIR},
+     * {@link ErrNo#EMFILE}, {@link ErrNo#ENFILE}, {@link ErrNo#ENOENT}, {@link ErrNo#ENOSPC},
+     * {@link ErrNo#ENXIO}, {@link ErrNo#EROFS}.
+     */
+    int open(String filename, int[] flags); // TODO what is mean with: "int flags[, mode_t mode]" https://www.gnu.org/software/libc/manual/html_mono/libc.html#Opening-and-Closing-Files
+
+    /**
+     * This function is obsolete and replaced by {@link #open(String, int[])}. <br>
+     * {@code creat(filename, mode)} is equivalent to: <br>
+     * {@code open(filename, O_WRONLY | O_CREAT | O_TRUNC, mode)}.
+     * <p>
+     * If on a 32 bit machine the sources are translated with _FILE_OFFSET_BITS == 64,
+     * returns a file descriptor opened in the large file mode which enables the file handling functions
+     * to use files up to 2^63 in size and offset from -2^63 to 2^63.
+     * This happens transparently for the user since all of the low-level file handling functions are equally replaced.
+     * @param filename
+     * @param mode
+     * @return
+     */
+    int creat(String filename, mode_t mode); // TODO how to handle mode_t type in Java?
```

Comment:
The javadoc says this function is obsolete, so you probably shouldn't map it.  That means you won't have to figure out how to map `mode_t`. :-) 

It's a variable length type, either `short` or `int` but I think a relic from the days of using 32-bit variables on 64-bit operating systems.   The "proper" way to handle that would be to have the native mappings get a `sizeof(mode_t)` and save it in a varable you can access like we do for `size_t`.  In reality you could probably map it to an `int` since nobody's on a 16-bit OS anymore.

Except it's obsolete so just delete this mapping.